main.go

```diff
@@ -138,7 +162,9 @@ func main() {
 	}()
 
 	env := envFromReader(rc)
```

Comment:
As mentioned in the general comment, this reverses the overriding logic, so I think it is better to put it behind a CLI flag.